tests/sklearn/test_sklearn_models.py

```diff
@@ -1710,6 +1710,10 @@ def test_p_error_simulation(
     # Check if model is linear
     is_linear_model = is_model_class_in_a_list(model_class, _get_sklearn_linear_models())
 
+    # Do not run the test for linear models since there is no PBS (i.e. p_error has no impact)
+    if is_linear_model:
```

Comment:
you should take the time to simplify the test here : we have no more reason to test linear models, so instead of having this `is_linear_model` and skip the test, you should simply change the 
```
@pytest.mark.parametrize("model_class, parameters", MODELS_AND_DATASETS)
```
to 
```
@pytest.mark.parametrize(
    "model_class, parameters",
    get_sklearn_linear_models_and_datasets() + get_sklearn_tree_models_and_datasets() + get_sklearn_neighbors_models_and_datasets()
)
```

we should keep `skip` for tests that will one day be re-enabled (after some fix or else)

Reply:
Indeed! I have not paid attention to the semantic my bad. 

You mean this right? No more linear models

```python
@pytest.mark.parametrize(
    "model_class, parameters",
    get_sklearn_neighbors_models_and_datasets()
    + get_sklearn_tree_models_and_datasets()
    + get_sklearn_neural_net_models_and_datasets(),
)
```

Reply:
Actually, the test makes sure linear models are not impacted by the p_error. We should keep that, right @RomanBredehoft ?

Reply:
ah yes indeed, but yes in that case don't skip it ! just modify the way we check in simulation , good catch

Reply:
Yes so this is done in #562.